bliss/metrics.py

```diff
@@ -188,8 +145,13 @@ def match_by_locs(true_locs, est_locs, slack=1.0):
     assert true_locs.shape[-1] == est_locs.shape[-1] == 2
     assert isinstance(true_locs, torch.Tensor) and isinstance(est_locs, torch.Tensor)
 
-    locs1 = true_locs.view(-1, 2)
-    locs2 = est_locs.view(-1, 2)
+    # reshape
+    locs1 = true_locs.view(-1, 2)  # 11x2 - coordinates for true light sources
+    locs2 = est_locs.view(-1, 2)  # 7x2 - coordinates for estimated light sources
+
+    # remove non-existent estimated/true light sources
+    locs1 = locs1[torch.abs(locs1).sum(dim=1) != 0]
+    locs2 = locs2[torch.abs(locs2).sum(dim=1) != 0]
```

Comment:
Can you use `n_sources` to remove off light sources rather than relying on locs summing to zero? Even if locs summing to zero works consistently, it's sort of hard from someone to understand

Reply:
Definitely, sounds good. Think it's also more logical to handle it like you suggested.